app/goal_routes.py

```diff
@@ -0,0 +1,132 @@
+from flask import Blueprint, request, make_response, abort, jsonify
+from app.models.goal import Goal
+from app.models.task import Task
+from app import db
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix = "/goals")
+
+# HELPER FUNCTION
+def validate_id(class_obj, id):
+    try:
+        object_id = int(id)
+    except:
+        abort(make_response({"details": f"{class_obj} {id} is invalid"}, 400))
+
+    query_result = class_obj.query.get(object_id)
+    if not query_result:
+        abort(make_response({"details": f"{class_obj} {id} is not found"}, 404))
+    
+    return query_result
+
+# CREATE RESOURCE
+@goals_bp.route("", methods = ["POST"])
```

Comment:
Based on PEP8 styling guide, "Don’t use spaces around the `=` sign when used to indicate a keyword argument, or when used to indicate a default value for an unannotated function parameter."
```suggestion
@goals_bp.route("", methods=["POST"])
```
[source link](https://peps.python.org/pep-0008/#other-recommendations)